sources/packages/api/src/services/restriction/student-restriction.service.ts

```diff
@@ -67,11 +68,16 @@ export class StudentRestrictionService extends RecordDataModelService<StudentRes
       query.andWhere("restrictionStudent.id = student.id");
     }
 
-    if (checkAll) {
-      query.andWhere("restrictions.actionType @> :restrictionActions");
+    if (restrictionActionType) {
+      query.andWhere("restrictions.actionType && :restrictionActionType");
```

Comment:
i didn't understand this? what is this for `query.andWhere("restrictions.actionType && :restrictionActionType");`?. we already have it in line 77

Reply:
and looks like the value of `restrictionActionType` is boolean, query is expecting an array

Reply:
Changed as we discussed.